client/src/main/java/org/apache/rocketmq/client/impl/producer/DefaultMQProducerImpl.java

```diff
@@ -326,11 +324,9 @@ public void run() {
                     try {
                         if (transactionCheckListener != null) {
                             localTransactionState = transactionCheckListener.checkLocalTransactionState(message);
-                        } else if (transactionListener != null) {
+                        } else {
                             log.debug("Used new check API in transaction message");
                             localTransactionState = transactionListener.checkLocalTransaction(message);
-                        } else {
```

Comment:
It's better not to delete this branch

Reply:
ping @Kvicii

Reply:
+1. Just leave the 'default' branch here may be better.